Automated Smoketest: :x:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 133e6456c40b312d8345d88327b829abe7ff42b4
Build: success
ECL Watch:  Rebuilding Site
Lint error(s): 
eclwatch/HPCCPlatformWidget.js: line 246, col 13, concise methods is available in ES6 (use esversion: 6) or Mozilla JS extensions (use moz).
eclwatch/LockDialogWidget.js: line 93, col 13, concise methods is available in ES6 (use esversion: 6) or Mozilla JS extensions (use moz).
2 errors
> eclwatch@1.0.0 test /mnt/disk1/home/vamosax/smoketest/PR-11730/HPCC-Platform/esp/src
> run-s lint
> eclwatch@1.0.0 lint /mnt/disk1/home/vamosax/smoketest/PR-11730/HPCC-Platform/esp/src
> jshint --config ./.jshintrc ./eclwatch
npm ERR! code ELIFECYCLE
> eclwatch@1.0.0 test /mnt/disk1/home/vamosax/smoketest/PR-11730/HPCC-Platform/esp/src
> run-s lint
> eclwatch@1.0.0 lint /mnt/disk1/home/vamosax/smoketest/PR-11730/HPCC-Platform/esp/src
> jshint --config ./.jshintrc ./eclwatch
npm ERR! errno 2
> eclwatch@1.0.0 test /mnt/disk1/home/vamosax/smoketest/PR-11730/HPCC-Platform/esp/src
> run-s lint
> eclwatch@1.0.0 lint /mnt/disk1/home/vamosax/smoketest/PR-11730/HPCC-Platform/esp/src
> jshint --config ./.jshintrc ./eclwatch
npm ERR! eclwatch@1.0.0 lint: `jshint --config ./.jshintrc ./eclwatch`
> eclwatch@1.0.0 test /mnt/disk1/home/vamosax/smoketest/PR-11730/HPCC-Platform/esp/src
> run-s lint
> eclwatch@1.0.0 lint /mnt/disk1/home/vamosax/smoketest/PR-11730/HPCC-Platform/esp/src
> jshint --config ./.jshintrc ./eclwatch
npm ERR! Exit status 2
> eclwatch@1.0.0 test /mnt/disk1/home/vamosax/smoketest/PR-11730/HPCC-Platform/esp/src
> run-s lint
> eclwatch@1.0.0 lint /mnt/disk1/home/vamosax/smoketest/PR-11730/HPCC-Platform/esp/src
> jshint --config ./.jshintrc ./eclwatch
npm ERR! 
> eclwatch@1.0.0 test /mnt/disk1/home/vamosax/smoketest/PR-11730/HPCC-Platform/esp/src
> run-s lint
> eclwatch@1.0.0 lint /mnt/disk1/home/vamosax/smoketest/PR-11730/HPCC-Platform/esp/src
> jshint --config ./.jshintrc ./eclwatch
npm ERR! Failed at the eclwatch@1.0.0 lint script.
> eclwatch@1.0.0 test /mnt/disk1/home/vamosax/smoketest/PR-11730/HPCC-Platform/esp/src
> run-s lint
> eclwatch@1.0.0 lint /mnt/disk1/home/vamosax/smoketest/PR-11730/HPCC-Platform/esp/src
> jshint --config ./.jshintrc ./eclwatch
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
> eclwatch@1.0.0 test /mnt/disk1/home/vamosax/smoketest/PR-11730/HPCC-Platform/esp/src
> run-s lint
> eclwatch@1.0.0 lint /mnt/disk1/home/vamosax/smoketest/PR-11730/HPCC-Platform/esp/src
> jshint --config ./.jshintrc ./eclwatch
npm ERR! A complete log of this run can be found in:
> eclwatch@1.0.0 test /mnt/disk1/home/vamosax/smoketest/PR-11730/HPCC-Platform/esp/src
> run-s lint
> eclwatch@1.0.0 lint /mnt/disk1/home/vamosax/smoketest/PR-11730/HPCC-Platform/esp/src
> jshint --config ./.jshintrc ./eclwatch
npm ERR!     /mnt/disk1/home/vamosax/.npm/_logs/2018-09-19T21_30_52_547Z-debug.log
> eclwatch@1.0.0 test /mnt/disk1/home/vamosax/smoketest/PR-11730/HPCC-Platform/esp/src
> run-s lint
> eclwatch@1.0.0 lint /mnt/disk1/home/vamosax/smoketest/PR-11730/HPCC-Platform/esp/src
> jshint --config ./.jshintrc ./eclwatch
ERROR: lint exited with 2.
npm ERR! Test failed.  See above for more details.

Lint warning(s): 
npm WARN Local package.json exists, but node_modules missing, did you mean to install?

Install hpccsystems-platform-community_7.0.0-rc2.el7.x86_64.rpm
Creating group hpcc ...\x1b[32m [   OK    ]  \x1b[0m 
 ... ( comment is too long, 102 lines, 4634 bytes, truncated) ... 
[Build error:](https://gist.github.com/5b4f623683fe37a0028eb7571c7e6f28)
Build log [Build log](https://gist.github.com/HPCCSmoketest/5b4f623683fe37a0028eb7571c7e6f28#file-result-18-09-19-17-29-55-log)

Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   40 sec (00:00:40) |  189 sec (00:03:09) |  171 sec (00:02:51) |   22 sec (00:00:22) |   44 sec (00:00:44) | 1488 sec (00:24:48) |   20 sec (00:00:20) | 1974 sec (00:32:54) |

```

--- a/esp/src/eclwatch/HPCCPlatformWidget.js
+++ b/esp/src/eclwatch/HPCCPlatformWidget.js
@@ -79,6 +79,7 @@ define([
 
             bannerContent: "",
             upgradeBar: null,
+            storage: null,
 
             postCreate: function (args) {
                 this.inherited(arguments);
@@ -234,6 +235,21 @@ define([
                 topic.subscribe("hpcc/monitoring_component_update", function (topic) {
                     context.checkMonitoring(topic.status);
                 });
+                this.storage = new ESPUtil.LocalStorage();
+                this.storage.on("storageUpdate", function(msg) {
+                    context._onUpdateFromStorage(msg)
+                });
+            },
+
+            _onUpdateFromStorage: function (msg){
+                var context = this;
+                if (msg.event.newValue === "logged_out") {
+                    window.location.reload();
+                } else if (msg.event.newValue === "Locked") {
+                    context._onShowLock();
+                } else if (msg.event.newValue === "Unlocked" || msg.event.oldValue === "Locked") {
+                    context._onHideLock();
+                }
             },
 
             initTab: function () {
@@ -439,12 +455,23 @@ define([
                 this.aboutDialog.hide();
             },
 
+            _onShowLock: function (evt) {
+                var LockDialog = new LockDialogWidget({});
+                LockDialog.show()
+            },
+
             _onLock: function (evt) {
                 var LockDialog = new LockDialogWidget({});
-                LockDialog.show();
+                LockDialog._onLock();
+            },
+
+            _onHideLock: function (evt) {
+                var LockDialog = new LockDialogWidget({});
+                LockDialog.hide();
             },
 
             _onLogout: function (evt) {
+                var context = this;
                 this.logoutConfirm.show();
                 query(".dijitDialogUnderlay").style("opacity", "0.5");
                 this.logoutConfirm.on("execute", function () {
@@ -455,6 +482,7 @@ define([
                             cookie("ECLWatchUser", "", { expires: -1 });
                             cookie("ESPSessionID" + location.port + " = '' ", "", { expires: -1 });
                             window.location.reload();
+                            context.storage.setItem("Status", "logged_out");
                         }
                     });
                 });

--- a/esp/src/eclwatch/LockDialogWidget.js
+++ b/esp/src/eclwatch/LockDialogWidget.js
@@ -13,6 +13,7 @@ define([
     "dojo/keys",
     "dojo/cookie",
     "dojo/topic",
+    "dojo/query",
 
     "dijit/registry",
     "dijit/form/Select",
@@ -21,6 +22,7 @@ define([
     "hpcc/_Widget",
     "src/Utility",
     "src/ws_account",
+    "src/ESPUtil",
 
     "dojo/text!../templates/LockDialogWidget.html",
 
@@ -32,9 +34,9 @@ define([
 
     "hpcc/TableContainer"
 
-], function (declare, lang, i18n, nlsHPCC, arrayUtil, dom, domForm, domClass, on, domStyle, xhr, keys, cookie, topic,
+], function (declare, lang, i18n, nlsHPCC, arrayUtil, dom, domForm, domClass, on, domStyle, xhr, keys, cookie, topic, query,
     registry, Select, CheckBox,
-    _Widget, Utility, WsAccount,
+    _Widget, Utility, WsAccount, ESPUtil,
     template) {
         return declare("LockDialogWidget", [_Widget], {
             templateString: template,
@@ -43,6 +45,7 @@ define([
 
             _width: "480px",
             lockDialogWidget: null,
+            storage: null,
 
             postCreate: function (args) {
                 this.inherited(arguments);
@@ -51,6 +54,8 @@ define([
                 this.unlockUserName = registry.byId(this.id + "UnlockUserName");
                 this.unlockPassword = registry.byId(this.id + "UnlockPassword");
                 this.unlockForm = registry.byId(this.id + "UnlockForm");
+                this.unlockStatus = dom.byId(this.id + "UnlockStatus");
+                this.storage = new ESPUtil.LocalStorage();
             },
 
             startup: function (args) {
@@ -58,15 +63,32 @@ define([
             },
 
             show: function (event) {
+                var context = this;
+                on(this.unlockPassword, "keypress", function (event) {
+                    if (event.key === "Enter") {
+                        context._onUnlock();
+                    }
+                });
+
                 this.unlockDialog.show();
                 domClass.add("SessionLock", "overlay");
-                dojo.removeClass(this.id + "UnlockDialog_underlay", "dijitDialogUnderlay _underlay");
                 this.unlockUserName.set("value", dojoConfig.username);
-                this._onLock();
+
+                topic.publish("hpcc/session_management_status", {
+                    status: "Locked"
+                });
             },
 
             hide: function (event) {
+                domClass.remove("SessionLock", "overlay");
                 this.unlockDialog.hide();
+                this.unlockDialog.destroyRecursive()
+                dojo.query(".dijitDialogUnderlayWrapper").forEach(function(node){
+                    dojo.destroy(node.id);
+                });
+                dojo.query(".unlockDialogToHide").forEach(function(node){
+                    dojo.destroy(node.id);
+                });
             },
 
             _onUnlock: function (event) {
@@ -80,21 +102,22 @@ define([
                             password: this.unlockPassword.get("value")
                         }
                     }).then(function (response) {
-                        var status = dom.byId("UnlockStatus");
 
                         if (response.UnlockResponse.Error === 0) {
-                            if (status.innerHTML !== "") {
-                                status.innerHTML = "";
+                            if (context.unlockStatus.innerHTML !== "") {
+                                context.unlockStatus.innerHTML = "";
                             }
-                            context.hide();
-                            context.unlockDialog.destroyRecursive();
                             domClass.remove("SessionLock", "overlay");
+                            context.unlockDialog.hide();
+                            context.unlockDialog.destroyRecursive();
                             topic.publish("hpcc/session_management_status", {
                                 status: "Unlocked"
                             });
                             cookie("Status", "Unlocked");
+                            context.storage.removeItem("Status");
+                            context.storage.setItem("Status", "Unlocked");
                         } else {
-                            status.innerHTML = response.UnlockResponse.Message;
+                            context.unlockStatus.innerHTML = response.UnlockResponse.Message;
                             cookie("Status", "Locked");
                         }
                     });
@@ -110,16 +133,23 @@ define([
                     }
                 });
 
-                xhr("esp/lock", {
-                    method: "post",
-                }).then(function(response){
-                    if (response) {
-                        topic.publish("hpcc/session_management_status", {
-                            status: "Locked"
-                        });
-                        cookie("Status", "Locked");
-                    }
-                });
+                if (cookie("Status") === "Unlocked") {
+                    xhr("esp/lock", {
+                        method: "post",
+                    }).then(function(response){
+                        if (response) {
+                            context.unlockDialog.show();
+                            domClass.add("SessionLock", "overlay");
+                            context.unlockUserName.set("value", dojoConfig.username);
+                            topic.publish("hpcc/session_management_status", {
+                                status: "Locked"
+                            });
+                            cookie("Status", "Locked");
+                            context.storage.removeItem("Status");
+                            context.storage.setItem("Status", "Locked");
+                        }
+                    });
+                }
             },
 
             init: function (params) {

--- a/esp/src/eclwatch/css/hpcc.css
+++ b/esp/src/eclwatch/css/hpcc.css
@@ -317,7 +317,7 @@ hr.dashedLine {
     left: 0;
     width: 100%;
     height: 100%;
-    z-index: 100;
+    z-index: 950;
     background-color: #1A9BD7;
 }
 .lockLogo{
@@ -615,6 +615,13 @@ hr.dashedLine {
     color: rgb(237, 28, 36);
 }
 
+.unlockStatus{
+    font-weight:bold;
+    font-size:1em;
+    color:rgb(218, 4, 0);
+    text-align: center;
+}
+
 table.miniSelect span {  
      width:20px;
 }
@@ -666,7 +673,7 @@ table.miniSelect span {
     padding-left:10px;
 }
 
-#UnlockStatus{
+.unlockStatus{
     font-weight:bold;
     font-size:1em;
     color:rgb(218, 4, 0);

--- a/esp/src/eclwatch/stub.js
+++ b/esp/src/eclwatch/stub.js
@@ -66,7 +66,7 @@ define([
             idleWatcher.on("idle", function () {
                 idleWatcher.stop();
                 var LockDialog = new LockDialogWidget({});
-                LockDialog.show();
+                LockDialog._onLock();
             });
             idleWatcher.start();
             monitorLockClick.unlocked();
@@ -95,6 +95,8 @@ define([
             topic.subscribe("hpcc/session_management_status", function (publishedMessage) {
                 if (publishedMessage.status === "Unlocked") {
                     monitorLockClick.unlocked();
+                } else  if (publishedMessage.status === "Locked") {
+                    monitorLockClick.locked();
                 }
             })
 

--- a/esp/src/eclwatch/templates/LockDialogWidget.html
+++ b/esp/src/eclwatch/templates/LockDialogWidget.html
@@ -1,15 +1,15 @@
 <span class="${baseClass}">
-    <div id="${id}UnlockDialog" title="${i18n.ReAuthenticate}" style="width: 480px; z-index: 9999;" data-dojo-props="closable:false, draggable: false, " data-dojo-type="dijit.Dialog">
+    <div id="${id}UnlockDialog" class="unlockDialogToHide" title="${i18n.ReAuthenticate}" style="width: 480px; z-index: 9999;" data-dojo-props="closable:false, draggable: false, " data-dojo-type="dijit.Dialog">
         <img src="${dojoConfig.urlInfo.resourcePath}/img/Loginlogo.png" class="lockLogo" alt="${i18n.HPCCSystems}"/>
         <div id="${id}UnlockForm" style="width:${_width}"  onsubmit="return false;" data-dojo-type="dijit.form.Form">
             <div data-dojo-props="cols:2" data-dojo-type="hpcc.TableContainer">
                 <input id="${id}UnlockUserName" title="${i18n.Username}:" name="username" colspan="2" style="width:90%;" data-dojo-props="trim: true, readonly: true" data-dojo-type="dijit.form.TextBox" />
                 <input id="${id}UnlockPassword" type="password" title="${i18n.Password}:" name="password" colspan="2" style="width:90%;" data-dojo-props="trim: true, readonly: false" data-dojo-type="dijit.form.ValidationTextBox" />
             </div>
-            <p id="UnlockStatus"></p>
+            <p id="${id}UnlockStatus" class="unlockStatus"></p>
         </div>
         <div class="dijitDialogPaneActionBar">
-            <button id="unlock" data-dojo-attach-event="onClick:_onUnlock" data-dojo-type="dijit.form.Button">${i18n.Unlock}</button>
+            <button id="${id}Unlock" data-dojo-attach-event="onClick:_onUnlock" data-dojo-type="dijit.form.Button">${i18n.Unlock}</button>
         </div>
     </div>
 </span>

--- a/esp/src/src/ESPUtil.ts
+++ b/esp/src/src/ESPUtil.ts
@@ -252,6 +252,33 @@ var GridHelper = declare(null, {
     }
 });
 
+export var LocalStorage = dojo.declare([Evented], {
+    constructor: function () {
+        var context = this;
+        if (typeof Storage !== void(0)) {
+            window.addEventListener('storage', function (event) {
+                context.emit('storageUpdate', {event});
+            });
+        } else {
+            console.log("Browser doesn't support multi-tab communication");
+        };
+    },
+
+    setItem: function (key, value) {
+        localStorage.setItem(key, value);
+    },
+    removeItem: function (key) {
+        localStorage.removeItem(key);
+    },
+    getItem: function (key) {
+        localStorage.getItem(key)
+    },
+    clear: function () {
+        localStorage.clear();
+    }
+});
+
+
 export var MonitorLockClick = dojo.declare([Evented], {
     unlocked: function () {
         var context = this;

